### DISCUSSION TITLE
How to avoid the behind sliceStore cover the front sliceStore?

### DISCUSSION THREAD

**Discussion by lightofyou** [pmndrs/zustand#925] [Q&A]
I follow the Beast Practices and split the store into seperate slices. Here is my code.

    // A.ts
    import { GetState, SetState } from "zustand";
    import { MyState } from "./index";
    
    export interface ISliceAStore {
      A: string
      C: number
    }
    
    type StoreFunction = (set: SetState<MyState>, get: GetState<MyState>) => ISliceAStore
    
    const createUserInfo: StoreFunction = (set, get) => ({
      A: 'A',
      C: 1
    });
    
    export default createUserInfo;
    

    // B.ts
    import { GetState, SetState } from "zustand";
    import { MyState } from "./index";
    
    export interface ISliceBStore {
      B: string
      C: number
    }
    
    type StoreFunction = (set: SetState<MyState>, get: GetState<MyState>) => ISliceBStore
    
    const createUserInfo: StoreFunction = (set, get) => ({
      B: 'B',
      C: 999
    });
    
    export default createUserInfo;
    

    // index.ts
    import create from "zustand";
    
    import createSliceAStore, { ISliceAStore } from "./A";
    import createSliceBStore, { ISliceBStore } from './B'
    
    export type MyState = ISliceAStore & ISliceBStore;
    
    const useStore = create<MyState>((set, get) => ({
      ...createSliceAStore(set, get),
      ...createSliceBStore(set, get)
    }));
    
    export default useStore;
    

The problem is that the C will be 999. The SliceBStore will cover the SliceAStore.

I want to ask how to avoid this like namespace?

Like [Pinia](https://pinia.vuejs.org/introduction.html#basic-example)

    const useCounterStore = defineStore('counter', {
      state: () => ({ count: 0 }),
      getters: {
        double: (state) => state.count * 2,
      },
      actions: {
        increment() {
          this.count++
        }
      }
    })

**Comment by dai-shi** [comment]
Hi, there's a long issue: #178 
Long story short, it's the field for 3rd-party libraries.
[zustand-lens](https://github.com/dhmk083/dhmk-zustand-lens) is the one which looks promising.
